src/test/java/com/exasol/adapter/dialects/saphana/util/IntegrationTestSetup.java

```diff
@@ -0,0 +1,189 @@
+
+import com.exasol.adapter.dialects.saphana.util.dbbuilder.HanaObjectFactory;
+import com.exasol.adapter.dialects.saphana.util.dbbuilder.HanaSchema;
+import com.exasol.bucketfs.Bucket;
+import com.exasol.bucketfs.BucketAccessException;
+import com.exasol.containers.ExasolContainer;
+import com.exasol.containers.ExasolService;
+import com.exasol.dbbuilder.dialects.Schema;
+import com.exasol.dbbuilder.dialects.exasol.*;
+import com.exasol.udfdebugging.UdfTestSetup;
+import com.github.dockerjava.api.model.ContainerNetwork;
+
+public class IntegrationTestSetup implements AutoCloseable {
+    private static final Logger LOGGER = Logger.getLogger(IntegrationTestSetup.class.getName());
+
+    private final HanaContainer<?> hanaContainer;
+    private final ExasolContainer<?> exasolContainer;
+    private final Connection exasolConnection;
+    private final Connection hanaConnection;
+    private final ExasolObjectFactory exasolFactory;
+    private final HanaObjectFactory hanaFactory;
+    private final AdapterScript adapterScript;
+    private final ConnectionDefinition connectionDefinition;
+    private int virtualSchemaCounter = 0;
+    private int hanaSchemaCounter = 0;
+
+    public IntegrationTestSetup(final HanaContainer<?> hana, final ExasolContainer<?> exasol)
+            throws NoDriverFoundException, SQLException {
+        this.hanaContainer = hana;
+        this.exasolContainer = exasol;
+        this.exasolConnection = this.createExasolConnection();
+        this.hanaConnection = DriverManager.getConnection(this.hanaContainer.getJdbcUrl(), hanaContainer.getUsername(),
+                hanaContainer.getPassword());
+        final ExasolObjectConfiguration.Builder builder = ExasolObjectConfiguration.builder();
+        final UdfTestSetup udfTestSetup = new UdfTestSetup(getTestHostIpFromInsideExasol(),
+                this.exasolContainer.getDefaultBucket(), this.exasolConnection);
+        builder.withJvmOptions(udfTestSetup.getJvmOptions());
+        this.exasolFactory = new ExasolObjectFactory(this.exasolConnection, builder.build());
+        this.hanaFactory = new HanaObjectFactory(this.hanaConnection);
+        final ExasolSchema exasolSchema = this.exasolFactory.createSchema(SCHEMA_EXASOL);
+
+        this.adapterScript = createAdapterScript(exasolSchema);
+        final String connectionString = "jdbc:sap://" + this.exasolContainer.getHostIp() + ":"
+                + this.hanaContainer.getSystemPort() + "/";
+        this.connectionDefinition = this.exasolFactory.createConnectionDefinition("HANA_CONNECTION", connectionString,
+                this.hanaContainer.getUsername(), this.hanaContainer.getPassword());
+    }
+
+    public static IntegrationTestSetup start() {
+        final HanaContainer<?> hana = new HanaContainer<>("saplabs/hanaexpress:2.00.061.00.20220519.1").withReuse(true);
+        final ExasolContainer<?> exasol = new ExasolContainer<>("7.1.17")
```

Comment:
Default version of the pinned ExasolContainer, no need for an additional pin, I think.

Reply:
Correct, but the default is 7.1.15, I want to test with the latest.